src/hooks/useMergedRefs.ts

```diff
@@ -1,57 +1,33 @@
-import { useCallback } from 'react';
+import type { RefCallback, Ref } from 'react';
 
 /**
- * NOTE: This was "borrowed" from reach-ui!
+ * Assign a value to a ref function or object.
  */
-
-/**
- * React.Ref uses the readonly type `React.RefObject` instead of
- * `React.MutableRefObject`, We pretty much always assume ref objects are
- * mutable (at least when we create them), so this type is a workaround so some
- * of the weird mechanics of using refs with TS.
- */
-export type AssignableRef<ValueType> =
-  | {
-      bivarianceHack(instance: ValueType | null): void;
-    }['bivarianceHack']
-  | React.MutableRefObject<ValueType | null>;
```

Comment:
Really just wanted to get rid of the lint rule suppression in here, but decided to simplify while I was at it. Should make it much more clear what's going on here.